packages/roosterjs-editor-types/lib/event/SelectionChangeEvent.ts

```diff
@@ -0,0 +1,50 @@
+import BasePluginEvent from './BasePluginEvent';
+import TableSelection from '../interface/TableSelection';
+import { PluginEventType } from '../enum/PluginEventType';
+import { SelectionRangeTypes } from '../enum/SelectionRangeTypes';
+import type { CompatibleSelectionRangeTypes } from '../compatibleEnum/SelectionRangeTypes';
+import type { CompatiblePluginEventType } from '../compatibleEnum/PluginEventType';
+
+/**
+ * Data of SelectionChangedEvent
+ */
+export interface SelectionChangedEventData {
+    /**
+     * Type of the selection
+     */
+    selectionType: SelectionRangeTypes | CompatibleSelectionRangeTypes;
+
+    /**
+     * The HTMLElement selected
+     */
+    selectedElement?: HTMLElement;
+
+    /**
+     * The range selected
+     */
+    range?: Range;
+
+    /**
+     * If is a table selection, the coordinates of the selection
+     */
+    coordinates?: TableSelection;
+
+    /**
+     * If the selection changed was caused by a keyboard event, the key that changed selection
+     */
+    keyboardKey?: number | string;
```

Comment:
I feel we don't need to pass the key info here. We should just know selection is changed, but no need to know the reason